src/tests/zf_apps/zfsink.c

```diff
@@ -265,18 +328,34 @@ static void monitor()
     ms += (end.tv_usec - start.tv_usec) / 1000;
     pkt_rate = (int) ((now_pkts - prev_pkts) * 1000 / ms);
     mbps = (int) ((now_bytes - prev_bytes) * 8 / 1000 / ms);
-    vlog("%10d %16d %16"PRIu64"\n", pkt_rate, mbps, now_pkts);
+    
+    pthread_mutex_lock(&printf_mutex);
+    printf("%10d %16d %16"PRIu64, pkt_rate, mbps, now_pkts);
+
+    if( cfg_intf_stats ) {
+      zf_stats_query(stack, stats_data, stats_collection, 1);
```

Comment:
It would be cleaner to move the zf_stats_query outside of the printf lock as it doesn't need it